packages/commerce-sdk-react/src/components/ShopperExperience/Component/index.tsx

```diff
@@ -24,17 +119,53 @@ const ComponentNotFound = ({typeId}: ComponentType) => (
  */
 export const Component = ({component}: ComponentProps) => {
     const pageContext = usePageContext()
-    const ComponentClass = pageContext?.components[component.typeId] || ComponentNotFound
+    const {jsxParserComponents} = pageContext
     const {data, ...rest} = component
+    const {code} = data || {}
+    const isEinsteinAssistedComponent = !!code
+    let instance = <ComponentNotFound {...rest} {...data} />
+
+    useEffect(() => {
```

Comment:
Is there a strong reason the eGPT component needs to be elevated to special status in this "container" component? Can it not just be another PD "ComponentClass" that is added to the `ComponentMap` of the page?

That would give you a few benefits:

- You can move this logic out of this file including the heavy `react-jsx-parser` dependency. It would only be brought in if the customer elects to add it to their component map and wouldn't affect everyones (or the default) bundle size. Right now this is brought in once they use this version of the SDK.
- Avoids adding new props to the `Page` wrapper like JsxParserComponents. Instead customers add the egpt assisted component where they are already used to
- Lets folks use the einstein assisted component directly (like in their own page designer implementations: like my own which is slightly different in it's wrappers)
  - Related to this you could, in theory if you wanted to, ship the eGPT component as a separate package instead of in commerce-sdk-react
- In theory customers can use the egpt editor in their own components that might have other attributes. So this would support that since they could dispatch the JSX part to be handled by the EGPT component.

Since you still need to know the "JsxParserComponents" instead of passing in the component class itself you can do a higher-order function create it. i.e. Thinking customers could initialize their page designer based page like this which would look very familiar 

```js
import {usePage} from '@salesforce/commerce-sdk-react'
import {Page, makeEinsteinAssistedComponent, DEFAULT_EGPT_JSX_COMPONENTS} from '@salesforce/commerce-sdk-react/components'

import {ImageTile, ImageWithText} from '../../page-designer/assets'

import {
    Carousel
} from '../../page-designer/layouts'

const PAGEDESIGNER_TO_COMPONENT = {
    'commerce_assets.photoTile': ImageTile,
    'commerce_assets.imageAndText': ImageWithText,
    'commerce_layouts.carousel': Carousel,
    'einstein.einsteinAssisted': makeEinsteinAssistedComponent({
        ...DEFAULT_EGPT_JSX_COMPONENTS,
        // Add missing table support to generative output
        TableContainer,
        Table,
        Thead,
        Tbody,
        Th,
        Td,
        Tr,
        TableCaption,

        SomeOtherComponent
    })
}

const PageViewer = () => {
    // ...
    return (
        <Box layerStyle={'page'}>
            <Page page={page} components={PAGEDESIGNER_TO_COMPONENT} />
        </Box>
    )
}
```